index.js

```diff
@@ -14,20 +14,20 @@ function SwiftClient(url, token) {
 util.inherits(SwiftClient, SwiftEntity);
 
 
-SwiftClient.create = function (url, username, password) {
+SwiftClient.create = function (url, account, version, username, password) {
   var _this = this;
 
   return requestp({
       method: 'GET',
-      uri: url,
+      uri:  urlHelper.combineUrl(urlHelper.combineUrl(url,"auth"),version),
       headers: {
         'x-auth-user': username,
         'x-auth-key': password
       },
       resolveWithFullResponse: true
     })
     .then(function (response) {
-      return new SwiftClient(url, response.headers['x-auth-token']);
+      return new SwiftClient(urlHelper.combineUrl(urlHelper.combineUrl(url, version), account), response.headers['x-auth-token']);
```

Comment:
the 'url' variable should actually be replaced with:

  response.headers['x-storage-url']